test-suite/unit-tests/src/utest.mli

```diff
@@ -10,3 +10,6 @@ val mk_bool_test : string -> string -> bool -> OUnit.test
    containing the tests; use __FILE__ for that purpose.
  *)
 val run_tests : string -> OUnit.test list -> unit
+
+(** output channel for the test log file *)
+val log_out_ch : out_channel ref
```

Comment:
if I get it right, this is a read-only ref, so please make it of type `unit -> out_channel` and name it `get_log_out_chan`.

Reply:
I removed the ref by adding an `open_log_out_ch` function and then passing the returned `out_channel` to `run_tests`.

Reply:
I plan to address the todo's related to the lexer soon after this PR is done.  Those changes aren't needed for the basic diff functionality, but may be helpful if others want to use this code show diffs in other parts of the code (e.g. in error messages showing expected and actual types).  And this PR has been pending for far too long.

Some of the todo's are my questions about OCaml that I'd appreciate reviewer feedback on since I'm new to the language.  Others are about non-essential "nice-to-have" changes that I may get to later, some of them driven by user feedback as they use the feature.

The one that says `(* todo: not empty! error message and return default *)` I'll make some adjustments.  The diff code is a) called within try blocks and b) can be turned off by the user, which means there's minimal risk of causing a critical bug.

What is your philosophy on todo's?  I could label them with something other than "todo" if that would help.  But I think it's helpful to capture some of these thoughts in comments in a findable way.

Reply:
I made a change for the todo I mentioned.  Now awaiting your further comments/questions.